lib/Common/Codex/Utf8Codex.cpp

```diff
@@ -37,7 +37,7 @@ namespace utf8
 
     inline bool ShouldFastPath(LPCUTF8 pb, LPCOLESTR pch)
     {
-        return (reinterpret_cast<size_t>(pb) & mAlignmentMask) == 0 || (reinterpret_cast<size_t>(pch) & mAlignmentMask) == 0;
+        return (reinterpret_cast<size_t>(pb) & mAlignmentMask) == 0 && (reinterpret_cast<size_t>(pch) & mAlignmentMask) == 0;
```

Comment:
Agree. (This change does not affect Scanner/Parser, only used in utf8~utf16 conversion routines, thus impact is relatively low.) Thanks for review!